newIDE/app/src/MainFrame/index.js

```diff
@@ -493,12 +504,27 @@ class MainFrame extends React.Component<Props, State> {
       });
   };
 
+  restartGuidelines = () => {
+    this.setState({
+      guidelinesOpen: !this.state.guidelinesOpen,
+      //TODO
+      //Ici je passe un index 0 et je le transmet en props à Guidelines component ou il y a plus simple ?
```

Comment:
So you can put an index to 0 here and give it to Guidelines as a prop yes. The only "gotcha" is that you have a duplication of the state: you have the "index" that is both in MainFrame and in Guidelines. You have two solutions as explained before for something else:

- either Guideline is "controlled", taking the current index as a prop. This means that when you want to change the index, you have to call a prop "changeIndex" that is doing a setState in MainFrame, to change index. (then, React will re-render MainFrame, so re-render Guidelines, so Guidelines will get the updated "index").

This way is usually the preferred in React. We say that the data "flows" from the top component (MainFrame) to the bottom (Guidelines) (you can consider the props as "pipes" flowing data from top components to bottom :))

* OR, you keep the current index in guidelines BUT you pass a `initialIndex` to Guidelines. In other words, you do an "uncontrolled" component. This being said, it's hard to "reset" anything, because by definition, reseting something is an "imperative" thing to do (rather than declarative). The issue is: how to know in Guidelines that you must reset the index to 0?? :/ The `initialIndex` was probably already 0 before!
So the actual solution is to use React "refs" to do imperative things, i.e: call a method on `Guidelines`. Like this:

1. Use `createRef` and `ref={...}` to store a reference to `Guidelines` inside `MainFrame`.
2. In Guidelines, create a new method called `reset` and that do `this.setState({ index: 0 })` (or anything else to go back to the first step).
3. In MainFrame, in restartGuidelines, you call the ref:

```
restartGuidelines = () => {
  // this._guidelines.current is a reference to your component. Beware, it can be `null` if for some reason it was not rendered on the screen by React.
  if (this._guidelines.current) this._guidelines.current.reset();
}
```

This "imperative" call to `reset` is usually not the best way to do things in React, but it can help, as before, to avoid "lifting the state" in MainFrame. It's ok here as we expect Guidelines to be more or less autonomous (MainFrame should not have any logic related to the guidelines).

Final note (explore this later): you can see that sharing component is easy in React, but sharing logic is hard. Because you have to "lift" the state everytime. So that's why React team created the "React hooks", to make it easy to share logic. But we don't use them in MainFrame.